Realm/RLMArrayLinkView.mm

```diff
@@ -205,6 +205,11 @@ - (NSUInteger)indexOfObject:(RLMObject *)object {
         @throw RLMException(@"RLMObject is no longer valid");
     }
 
+    // check that object types align
+    if (![_objectClassName isEqualToString:object.objectSchema.className]) {
+        @throw RLMException(@"Object type does not match RLMArray");
```

Comment:
`[NSString stringWithFormat:@"Object of type (%@) does not match RLMArray type (%@)", object.objectSchema.className, _objectClassName]`